extensions/data-plane/data-plane-azure-storage/src/main/java/org/eclipse/edc/connector/dataplane/azure/storage/DestinationBlobName.java

```diff
@@ -16,34 +16,53 @@
 
 import org.eclipse.edc.util.string.StringUtils;
 
+import java.util.Optional;
+
 /**
  * Utility class responsible for determining the name under which a file will be saved.
  */
 public class DestinationBlobName {
 
-    private final String folderName;
-    private final String blobName;
+    private final Optional<String> folderName;
+    private final Optional<String> blobName;
+    private static final String PARTNAME_VALIDATION_MESSAGE = "partName cannot be null or blank when blobName is empty or not provided.";
+
 
     public DestinationBlobName(String blobName, String folderName) {
 
-        this.blobName = blobName;
-        this.folderName = folderName;
+        this.blobName = Optional.ofNullable(blobName);
+        this.folderName = Optional.ofNullable(folderName);
     }
 
     /**
      * Resolves the name under which a resource should be saved based on its part name and size.
      *
-     * @param partName The name of the resource part.
+     * @param partName  The name of the resource part.
      * @param partsSize The size of the resource part.
      * @return A String representing the resolved name for the resource.
      */
     public String resolve(String partName, int partsSize) {
 
-        var name = (partsSize == 1 && !StringUtils.isNullOrEmpty(this.blobName)) ? this.blobName : partName;
-        if (!StringUtils.isNullOrEmpty(this.folderName)) {
-            name = this.folderName.endsWith("/") ? this.folderName + name : this.folderName + "/" + name;
+        var sb = new StringBuilder();
+        if (blobName.isEmpty() || blobName.get().isBlank()) {
+            if (StringUtils.isNullOrBlank(partName)) {
+                throw new IllegalArgumentException(PARTNAME_VALIDATION_MESSAGE);
+            }
+        }
+        if (folderName.isPresent() && !folderName.get().isBlank()) {
+            if (folderName.get().endsWith("/")) {
+                sb.append(folderName.get());
+            } else {
+                sb.append(folderName.get()).append("/");
+            }
         }
-        return name;
+        if (partsSize == 1 && blobName.isPresent() && !blobName.get().isBlank()) {
+            sb.append(blobName.get());
+        } else {
+            sb.append(partName);
+        }
+
+        return sb.toString();
```

Comment:
apart from the "optional" thing I don't understand why to add such complexity in this method.
`Optional` can be used only if it make things easier, but in this case `isNullOrEmpty` already did the job quite well, so there's no real need to change the previous implementation.